secure_message/common/utilities.py

```diff
@@ -78,30 +79,44 @@ def process_paginated_list(paginated_list, host_url, user, message_args, endpoin
 
 
 def add_to_details(messages):
-    """Adds a @msg_to key every message in a list of messages.
+    """Adds a @msg_to key to every message in a list of messages.
     Every msg_to uuid is resolved to include details of the user.
 
     If the call for the internal user id fails, an exception will be thrown.
     If the external user id cannot be found in the list that we got from the party service.  There
     won't be a @msg_to value returned in the payload.  The API documentation notes that these elements
     aren't guaranteed to be provided so we're not breaking the contract by doing this.
+
+    Note:28th Feb 2019. Try except blocks added to try and identify cause of IndexError seen in splunk logs
+    No functionality added, just log and raise.
     """
     external_user_details = {}
-    for user in party.get_users_details(get_external_user_uuid_list(messages)):
+
+    try:
+        external_users = get_external_user_uuid_list(messages)
+    except IndexError:
+        logger.exception(f"Exception getting external user uuids for messages:'{json.dumps(messages)}'")
```

Comment:
This is probably a bad idea.  What's inside of those messages?  We're not allowed to log any personally identifiable information and I think this might contain some of that bad stuff

Reply:
Good catch , the problem is I suspect that missing data may be causing this some of which is personally identifiable . This needs a rethink..